Library/Usage.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml;
+
+namespace Recurly
+{
+    public class Usage : RecurlyEntity {
+
+        public enum Type
+        {
+            Price,
+            Percentage
+        }
+
+        public int? UnitAmountInCents { get; set; }
+        public float? UsagePercentage { get; set; }
+        public int Amount{ get; set; }
+        public String MerchantTag { get; set; }
+        public Type UsageType { get; set; }
+        public DateTime? UsageTimestamp { get; set; }
```

Comment:
Shouldn't the UsageTimestamp be always required? I think it should be. At least, it appears as required in the documentation https://dev.recurly.com/v2.2/docs/log-usage
About RecordingTimestamp not sure if it should be mandatory also.